### PR TITLE
Fix tar name to directory translation

### LINKED ISSUE #23635: installPackagePath https path regression
```
ic manifest generate --set installPackagePath=https://github.com/istio/istio/releases/download/1.6.0-beta.0/istio-1.6.0-beta.0-linux-amd64.tar.gz
```
fails because the extracted dir has path /tmp/istio-install-packages/istio-1.6.0-beta.0/ instead of /tmp/istio-install-packages/istio-1.6.0-beta.0-linux/ as expected.

### PR DESCRIPTION
**Pull request by ostromart**
Fixes https://github.com/istio/istio/issues/23635.

### CODE REVIEW

File: operator/pkg/helm/urlfetcher.go [hunk 1]
```diff
@@ -123,13 +123,19 @@ func DownloadTo(srcURL, dest string) (string, error) {
...
 	}
 	ver, err := version.NewVersionFromString(fv[1])
 	if err != nil {
 		return "", nil, err
 	}
-	// get rid of the suffix
-	return fn[:strings.LastIndex(fn, "-")], ver, nil
+	// get rid of the suffix like -linux-arch or -osx
+	if fv[fvl-2] == "linux" {
+		// linux names also have arch appended, trim this off
```
Comment (howardjohn) [inline operator/pkg/helm/urlfetcher.go:136]: only in new versions, do we need to support both?

**Review by howardjohn — COMMENTED** [review]
This is pretty hard to follow, can we have a test? seems easily testable

File: operator/pkg/helm/urlfetcher.go (hunk 1 above)
Comment (ostromart) [reply operator/pkg/helm/urlfetcher.go:136]: i think this does support both. in the old version, there's no -arch suffix so the if test fails and nothing is trimmed.

**Comment by ostromart** [comment]
@howardjohn yes and no on how easy it is to test. i can certainly put up a shell script that tests the command as part of this pr. but we'd have to figure out the right place to put it - it would be fetching files from real URLs so presubmit is not a good place for it, probably some release qual test? i'm not sure a UT that tests the function logic would be super useful here but i can put one in too if you think it will be helpful.

**Review by howardjohn — APPROVED** [review]
I meant unit testing this method so we don't need to discuss if it supports the old format or doesn't - the test would verify that. Lots of string parsing and stuff like this is high value testing in my opinion
